src/CollisionDetection.hs

```diff
@@ -1,55 +1,77 @@
-  --     recXP2 = -paddlesDistance
-  --     recYP2 = player2 game
-  --
-  --     -- Player A's paddle's left bottom corner (needed for collision math)
-  --     rectCornerXP1 = recXP1 -paddleWidth / 2
-  --     rectCornerYP1 = recYP1 -paddleHeight / 2
-  --
-  --     -- Player 2's paddle's left bottom corner (needed for collision math)
-  --     rectCornerXP2 = recXP2 -paddleWidth / 2
-  --     rectCornerYP2 = recYP2 -paddleHeight / 2
-  --
-  --     deltaXP1 = ballX - max rectCornerXP1 (min ballX (rectCornerXP1 + paddleWidth))
-  --     deltaYP1 = ballY - max rectCornerYP1 (min ballY (rectCornerYP1 + paddleHeight))
-  --
-  --     deltaXP2 = ballX - max rectCornerXP2 (min ballX (rectCornerXP2 + paddleWidth))
-  --     deltaYP2 = ballY - max rectCornerYP2 (min ballY (rectCornerYP2 + paddleHeight))
+  data CollisionSide =
+    TopSide | BottomSide | LeftSide | RightSide
+    deriving Show
 
   data WallCollisionType =
     TopWall | BottomWall | LeftWall | RightWall
 
-  -- | Given position and radius of the ball, return whether a collision occurred on the wall.
+  -- | Given position and radius of the ball, return whether
+  --   a collision occurred on the wall.
   wallsCollision :: Position                -- ^ ball position
                 -> Radius                   -- ^ ball radius
-                -> Float                    -- ^ window width
-                -> Float                    -- ^ window height
+                -> Width                    -- ^ game width
+                -> Height                    -- ^ game height
                 -> Maybe WallCollisionType  -- ^ collision with the walls?
   wallsCollision (x, y) radius width height
           | y + radius >=  height / 2 = Just TopWall    -- ^ Top wall
           | y - radius <= -height / 2 = Just BottomWall -- ^ Bottom wall
           | x - radius <= -width / 2 = Just LeftWall    -- ^ Left wall
           | x + radius >=  width / 2 = Just RightWall   -- ^ Right wall
           | otherwise = Nothing
+
+  -- | Given position and radius of the ball, return whether
+  --   a collision occurred on the brick
+  bricksCollision :: Position      -- ^ ball position
+                 -> Radius        -- ^ ball radius
+                 -> [Brick]         -- ^ bricks List
+                 -> (Maybe CollisionSide, [Brick])   -- ^ CollisionSide
+  bricksCollision ballL ballR bricks = go ballL ballR bricks (Nothing,[])
+            where go :: Position -> Radius -> [Brick] -> (Maybe CollisionSide, [Brick]) -> (Maybe CollisionSide, [ Brick ])
+                  go _ _ [] resp = resp
+                  go ballL ballR (b:bs) (col, brickLts) =
+                      case collision of
+                        Nothing   -> go ballL ballR bs (col, b:brickLts)
+                        _ -> (collision, brickLts ++ bs)
+                      where
+                        collision = rectangleCircleCollision ballL ballR (brickX - brickWidth / 2, brickY - brickHeight / 2) brickWidth brickHeight
```

Comment:
You should try limiting line length to 80 characters at most. Don't hesitate to break long expressions into multiple lines.